modules/control/common/leadlag_controller.cc

```diff
@@ -0,0 +1,91 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/control/common/leadlag_controller.h"
+
+#include <cmath>
+
+#include "cyber/common/log.h"
+
+namespace apollo {
+namespace control {
+
+double LeadlagController::Control(const double error, const double dt) {
+  if (dt <= 0) {
+    AWARN << "dt <= 0, will use the last output, dt: " << dt;
+    return innerstate_ * kn1_ + previous_innerstate_ * kn0_;
+  }
+  double output = 0;
+
+  innerstate_ = (error - previous_innerstate_ * kd0_) / kd1_;
+  // calculate the inner (intermedia) state under the Direct form II
+  // of the Lead / Lag compensator factorization
+  if (innerstate_ > innerstate_saturation_high_) {
+    innerstate_  = innerstate_saturation_high_;
```

Comment:
Always remove redundant whitespaces. Please also make sure "apollo.sh build" and "apollo.sh lint" pass.